conda_build/api.py

```diff
@@ -262,12 +286,12 @@ def build(
 
 
 def test(
-    recipedir_or_package_or_metadata,
-    move_broken=True,
-    config=None,
-    stats=None,
+    recipedir_or_package_or_metadata: str | os.PathLike | Path | MetaData,
+    move_broken: bool = True,
+    config: Config | None = None,
+    stats: dict | None = None,
```

Comment:
Can we add type annotations to `dict[]`? Even if it is something like `dict[str, Any]`, that'd be an improvement.

Reply:
I see that `stats` comes up in a few places. We could use a type alias so we don't have to repeat ourselves so much.

Reply:
Added a basic `StatsDict` type for now. Will need to continue iterating on it as we work our way through the code.